x-pack/filebeat/input/cel/input.go

```diff
@@ -961,19 +1005,64 @@ func newClient(ctx context.Context, cfg config, log *logp.Logger, reg *monitorin
 	}
 
 	if cfg.Auth.OAuth2.isEnabled() {
-		authClient, err := cfg.Auth.OAuth2.client(ctx, c)
+		c, err = cfg.Auth.OAuth2.client(ctx, c)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, err
 		}
-		return authClient, trace, nil
 	}
 
 	c.Transport = userAgentDecorator{
 		UserAgent: userAgent,
 		Transport: c.Transport,
 	}
 
-	return c, trace, nil
+	return c, trace, otelMetrics, nil
+}
+
+func CreateOTELMetrics(ctx context.Context, cfg config, log *logp.Logger, env v2.Context, tripper http.RoundTripper) (*OTELCELMetrics, *otelhttp.Transport, error) {
+	resource := resource.NewWithAttributes(
+		semconv.SchemaURL, getResourceAttributes(env, cfg)...,
+	)
+
+	log.Infof("created cel input resource %s", resource.String())
+	exporter, exporterType, err := otel.GetGlobalMetricsExporterFactory().GetExporter(ctx, true)
+	if err != nil {
+		log.Errorw("failed to get exporter", "error", err)
+	}
+	log.Infof("created OTEL cel input exporter %s for input %s", exporterType, env.IDWithoutName)
+	return NewOTELCELMetrics(log, *resource, tripper, exporter)
+}
+
+func getResourceAttributes(env v2.Context, cfg config) []attribute.KeyValue {
+	attrs := []attribute.KeyValue{
+		semconv.ServiceInstanceID(env.IDWithoutName),
+		attribute.String("package.name", cfg.GetPackageData("name")),
+		attribute.String("package.version", cfg.GetPackageData("version")),
+		attribute.String("package.data_stream", cfg.DataStream),
+		attribute.String("agent.version", env.Agent.Version),
+		attribute.String("agent.id", env.Agent.ID.String()),
+	}
+
+	attributes := os.Getenv("OTEL_RESOURCE_ATTRIBUTES")
+	if attributes == "" {
+		return attrs
+	}
+
+	seen := make(map[attribute.Key]bool)
+
```

Comment:
```suggestion
```
possibly also in the same stanza as the pairs range since this is entirely related to the use there.